client/src/handlers/handlers.ts

```diff
@@ -14,6 +14,189 @@ export function makeHandleRunJob (
   }
 }
 
+let flixFileLocation
+const FLIX_GLOB_PATTERN = '**/*.flix'
+
+/**
+ * It sets the path of flix.jar to a local variable `flixFileLocation` in file `handlers.ts`.
+ * 
+ * @param Filename String (path of the flix.jar)
+ * 
+ * @return void
+*/
+export function setFlixFileName(Filename:String) {
+    flixFileLocation = Filename
+}
+
+/**
+ * returns a count of total active terminals with prefix name `flix`.
+ * 
+ * @return number
+*/
+function countFlixTerminals() {
+    const activeTerminals = vscode.window.terminals
+    let count = 0
+    for (let i = 0; i < activeTerminals.length; i++) {
+        const element = activeTerminals[i];
+        if(element.name.substring(0, 4) == `flix`)
+            count+=1
+    }
+    return count
+}
+
+/**
+ * returns an active terminal with prefix name `flix`.
+ * 
+ * If not any active terminal with prefix name `flix`, it creates a new terminal with name `flix`.
+ *
+ * @return vscode.Terminal
+*/
+function ensureFlixTerminal() {
+    const activeTerminals = vscode.window.terminals
+    for (let i = 0; i < activeTerminals.length; i++) {
+        const element = activeTerminals[i];
+        if(element.name.substring(0, 4) == `flix`)
+            return element
+    }
+    return vscode.window.createTerminal(`flix`);
+}
+
+
+/**
+ * returns an new active terminal with prefix name `flix`.
+ * 
+ * If not any active terminal with prefix name `flix`, it creates a new terminal with name `flix` and returns it.
+ *
+ * If there are already `n` active terminals exist with prefix name `flix`, it creates a new terminal with name `flix n+1`
+ *
+ * @return vscode.Terminal
+*/
+function ensureNewFlixTerminal() {
+    const count = countFlixTerminals()
```

Comment:
Would it make sense to instead maintain an integer ourselves?

Reply:
Assume the user creates 5 terminals (flix-0, flix-1, ...., flix-4) and then deletes all. Now if we maintain an integer, the next created terminal will be flix-5 although there are no active flix terminals.

Reply:
What if the user creates 2 terminals, then deletes the first one, then creates a new one? Then there would be two terminals called `flix-1`, right?

Reply:
> What if the user creates 2 terminals, then deletes the first one, then creates a new one? Then there would be two terminals called `flix-1`, right?

yes

Reply:
So, If I start closing some terminals it could be pretty hard to know which terminal the program is currently running in?

Reply:
> So, If I start closing some terminals it could be pretty hard to know which terminal the program is currently running in?

:thumbsup: updated. maintained an integer that keeps track of the total number of flix terminals.